### PR TITLE
Assignment 2 - Alex L & Tyler Rose

### PR DESCRIPTION
**Pull request by Xela-19**
https://thankful-island-092b59a1e.3.azurestaticapps.net/

### CODE REVIEW

File: wordle-web/src/views/WordleView.vue [hunk 1]
```diff
@@ -1,55 +1,54 @@
...
 console.log(game.secretWord)
 
-onMounted(() => {
-  window.addEventListener('keyup', keyPress)
-})
-onUnmounted(() => {
-  window.removeEventListener('keyup', keyPress)
-})
-
 function checkGuess() {
```
Comment (grahamna) [inline wordle-web/src/views/WordleView.vue:31]: I think you should check for length of guess, and probably do some regex operations to filter non-letter chars here.

File: wordle-web/src/views/HomeView.vue [hunk 2]
```diff
@@ -1,9 +1,52 @@
...
+export default {
+  data: () => ({
+    showImage: false
+  })
+}
+</script>
+
+<template id="body">
+  <h1 id="h1">
+    Welcome to . . . <v-btn color="secondary" @click="showImage = !showImage">CLICK ME</v-btn>
```
Comment (grahamna) [inline wordle-web/src/views/HomeView.vue:16]: there isn't an obvious way to make my way from the welcome page to the wordle page from here. I suggest you make a btn appear which takes you to the game post logo reveal.

File: wordle-web/src/components/AvailableWordsButton.vue [hunk 3]
```diff
@@ -0,0 +1,82 @@
+<template>
+  <v-menu :close-on-content-click="false" v-model:model-value="showMenu">
```
Comment (grahamna) [inline wordle-web/src/components/AvailableWordsButton.vue:2]: I think there's supposed to be a hover element in this component, to make sure/highlight the word you're hovering over?

File: wordle-web/src/components/KeyBoard.vue [hunk 4]
```diff
@@ -38,11 +87,54 @@ const keyboardLetters = computed(() => {
...
+  if (event.key === 'Enter') {
+    emits('checkGuess')
+  } else if (event.key === 'Backspace') {
+    emits('backspace')
+  } else if (event.key.length === 1 && event.key !== ' ') {
+    emits('letterClick', new Letter(event.key.toLowerCase()))
+  }
+}
+
+function playClick() {
```
Comment (grahamna) [inline wordle-web/src/components/KeyBoard.vue:127]: I like this. This should get extra credit

File: wordle-web/src/components/SettingsMenu.vue [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        <v-toolbar color="primary">
+          <v-toolbar-title class="flex text-center">
+            <span class="text-h4"> Settings </span>
+          </v-toolbar-title>
+        </v-toolbar>
+        <v-card-text>
+          <v-card-title class="text-h5 pb-8"> Theme: </v-card-title>
+
+          <v-btn variant="tonal" color="#9575CD" @click="setDarkTheme">Dark</v-btn> &nbsp;
+          <v-btn variant="tonal" color="#4CAF50" @click="setLightTheme">Light</v-btn> &nbsp;
```
Comment (grahamna) [inline wordle-web/src/components/SettingsMenu.vue:41]: I believe that light and dark themes were supposed to be one button which toggles between light and dark modes. I could be wrong.

Comment (Xela-19) [reply wordle-web/src/components/SettingsMenu.vue:41]: I think you're right, I just wasn't sure exactly how to do that, but I am going to look into it.

File: wordle-web/src/components/AvailableWordsButton.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (Xela-19) [reply wordle-web/src/components/AvailableWordsButton.vue:2]: It does get darker when you hover over it - both the button to open the menu and the words themselves - so I'm not quite sure what you're referring to. If I'm understanding it correctly, it does this by default, so why add more code that would be saying the same thing? I'm pretty new to this, so if I'm missing something please explain :)

[the remark this answers is not in the document]
Comment (grahamna) [reply wordle-web/src/components/AvailableWordsButton.vue:2]: Ahh, you're right. I must of had my brightness too dim or had glare on my screen to notice it. My bad.

**Review by mmwoodfo — COMMENTED** [review]
Bug report, as I type my name it enters into the keyboard. :)
image


Also, when I type, the input is duplicated

Lastly - I don't think your available words works quite right... but Grant mentioned on Tuesday this was a difficult problem, so I think it works pretty well for your first shot at it.

**Pull request closed without merging**